### ISSUE TITLE
overloading methods in class

### ISSUE THREAD

**Issue by xSpecialFoodx** [microsoft/pylance-release#5378]
i've discovered a bug in 2023.12.1 that didn't exist in 2023.11.10.
if i have a class with 2 methods overloaded or more it doesn't suggest the parameters when i do a( for example.
```py
from typing import overload

class a:
  @overload
  def __init__(self, x: int):
    ...

  @overload
  def __init__(self, x: float):
    ...

  def __init__(self, x):
    return 3
```
a( gives me no suggestions.
with 1 overload it works.
if the overloads are for functions at top level and not class methods it works with multiple overloads though.
python 3.11.6.

edit:
i've managed to find that it works on 11.11 and downwards, and stops to work at 11.12 and upwards.

**Comment by debonte** [comment]
Looks like this is fixed in Pylance prerelease build 2023.12.104, which we just shipped this morning.

**Comment by debonte** [comment]
Duplicate issue in pyright: https://github.com/microsoft/pyright/issues/7093

**Comment by xSpecialFoodx** [comment]
> Looks like this is fixed in Pylance prerelease build 2023.12.104, which we just shipped this morning.

what a coincidence, that's nice to hear.
